src/main/java/net/minecraftforge/client/gui/config/CategoryConfigScreen.java

```diff
@@ -0,0 +1,294 @@
+        if (translationKey.equals(title.getString()))
+            return new StringTextComponent(fallback);
+        return title;
+    }
+
+    static Collection<? extends ITextProperties> translateCommentWithFallback(@Nullable String translationKey, @Nullable String fallbackComment) {
+        // Need to handle line breaks (\n) properly
+        if (translationKey != null) {
+            TranslationTextComponent comment = new TranslationTextComponent(translationKey);
+            if (!translationKey.equals(comment.getString())) {
+                comment.func_240701_a_(TOOLTIP_COMMENT_COLOR); // mergeStyle
+                return Collections.singletonList(comment);
+            }
+        }
+        if (fallbackComment == null)
+            return Collections.emptyList();
+        return Arrays.stream(fallbackComment.split("\n"))
+            .map(StringTextComponent::new)
+            .map(s -> s.func_240701_a_(TOOLTIP_COMMENT_COLOR)) // mergeStyle
+            .collect(Collectors.toList());
+    }
+
+    public static List<ITextProperties> createTooltip(ITextComponent title, @Nullable String commentTranslationKey, @Nullable String comment) {
+        List<ITextProperties> tooltip = new LinkedList<>();
+        tooltip.add(title.func_230532_e_().func_240701_a_(TOOLTIP_TITLE_COLOR));
+        tooltip.addAll(translateCommentWithFallback(commentTranslationKey, comment));
+        return tooltip;
+    }
+
+    @Override
+    protected ConfigElementList makeConfigElementList() {
+        return new CategoryConfigElementList(this, field_230706_i_, categoryInfo);
+    }
+
+    public interface ConfigCategoryInfo {
+
+        static ConfigCategoryInfo of(Supplier<Collection<String>> elements, Function<String, Object> getValue, Function<String, Object> getSpec) {
+            return of(elements, getValue, getSpec, $ -> null);
+        }
+
+        static ConfigCategoryInfo of(Supplier<Collection<String>> elements, Function<String, Object> getValue, Function<String, Object> getSpec, Function<String, String> getCategoryComment) {
+            return new ConfigCategoryInfo() {
+
+                @Override
+                public Collection<String> elements() {
+                    return elements.get();
+                }
+
+                @Override
+                public Object getValue(String key) {
+                    return getValue.apply(key);
+                }
+
+                @Override
+                public Object getSpec(String key) {
+                    return getSpec.apply(key);
+                }
+
+                @Override
+                public String getCategoryComment(String key) {
+                    return getCategoryComment.apply(key);
+                }
+            };
+        }
+
+        Collection<String> elements();
+
+        Object getValue(String key);
+
+        Object getSpec(String key);
+
+        /** Special case categories because reasons */
```

Comment:
Would be nice to have a little more understandable comment than `because reasons`

Reply:
Ya true.
In Forge's configs, all values are stored as `ConfigValue<T>`s with their corresponding `ValueSpec` stored in a sibling map.
Except for categories, they're stored as raw `SimpleConfig`s (`SimpleCommentedConfig`s after my changes) with no corresponding `ValueSpec`.
This means that comments are handled separately (in a map `levelComments` before my changes and that translation keys are not possible).
I'll add this explanation into the code.
It was initially so vague because I want to talk through my changes with someone (that improve comments) and also talk about how translation keys can be implemented and how to also allow them to store extra data in the future.